base/dataset.go

```diff
@@ -27,11 +27,18 @@ type loader struct {
 	fs qfs.Filesystem
 }
 
-// LoadDataset fetches, derefernces and opens a dataset from a reference
-// implements the dsfs.Loader interface
-func (l loader) LoadDataset(ctx context.Context, ref dsref.Ref, source string) (*dataset.Dataset, error) {
-	if source != "" {
-		return nil, fmt.Errorf("only local datasets can be loaded")
+// LoadDataset is not implemented for base
+func (l loader) LoadDataset(ctx context.Context, refstr string) (*dataset.Dataset, error) {
+	return nil, fmt.Errorf("not implemented")
+}
```

Comment:
This base implementation of Loader is only used for tests (sql and transform), which are then defining their own wrapper around it (what used to be ParseAndResolveFunc). It may be possible to remove it instead, but I haven't spent time yet to figure out how much test cleanup that requires.